libraries/ms-common/inc/blink_event_generator.h

```diff
@@ -0,0 +1,42 @@
+#pragma once
+
+// Takes an event ID with a data field and generates events toggling the data field between 1 and 0.
+// Requires interrupts, soft timers, and the event queue to be initialized.
+
+#include "event_queue.h"
+#include "soft_timer.h"
+
+typedef enum {
+  BLINKER_STATE_OFF = 0,
+  BLINKER_STATE_ON,
+  NUM_BLINKER_STATES,
+} BlinkerState;
+
+typedef struct {
+  uint32_t interval_us;
+  BlinkerState default_state;  // if not specified, the default state is BLINKER_STATE_OFF
```

Comment:
Nit. comment is a bit long. 
Something like "defaults to BLINKER_STATE_OFF" is enough :)